fixtures/_fixtures/monkeypatch.py

```diff
@@ -17,27 +17,70 @@
     'MonkeyPatch'
     ]
 
+import functools
 import sys
 import types
 
 from fixtures import Fixture
 
 
-def _setattr(obj, name, value):
-    """Handle some corner cases when calling setattr.
+_class_types = (type, )
+if getattr(types, 'ClassType', None):
+    # Python 2 has multiple types of classes.
+    _class_types = _class_types + (types.ClassType,)
 
-    setattr transforms a function into instancemethod, so where appropriate
-    value needs to be wrapped with staticmethod().
+
+def _coerce_values(obj, name, new_value, sentinel):
+    """Handle value coercion for static and classmethods.
+
+    setattr transforms a function into an instancemethod when set on a class.
+    This checks if the attribute to be replaced is either and wraps new_value
+    if necessary. This also checks getattr(obj, name) and wraps it if necessary
+    since the staticmethod wrapper isn't preserved.
```

Comment:
Might also be worth documenting the return values from this function? It seems it returns the (new_value, old_value) unless the thing we're patching doesn't exist already, in which case we return (new_value, sentinel)?